core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -760,6 +760,24 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
   def delayBy[F2[x] >: F[x]: Temporal](d: FiniteDuration): Stream[F2, O] =
     Stream.sleep_[F2](d) ++ this
 
+  /** Ensure that the stream always emits elements by defining a maxIdle duration.
+    * In other words, the stream will emit an element when it hasn't emitted any since the maximum time specified.
+    */
+  def keepAlive[F2[x] >: F[x]: Temporal, O2 >: O](
+      maxIdle: FiniteDuration,
+      heartbeat: O2
```

Comment:
Better to be a function.

Reply:
You suggest, update to : `heartbeat: => O2` ?

Reply:
Yes, a user may provide different heart beat messages, iirc, pekko akka stream is act like that.

Reply:
I thought about it but couldn't figure out how to make a Pull.output from a F effect.

Reply:
`Pull.eval(o2).flatMap(o => Pull.output1(o))`